python/mxnet/executor_manager.py

```diff
@@ -283,8 +283,14 @@ def update_metric(self, metric, labels):
         """ Update evaluation metric with label and current outputs """
         for texec, islice in zip(self.train_execs, self.slices):
             n = int(texec.outputs[0].shape[0] / (islice.stop - islice.start))
-            new_slice = slice(islice.start * n, islice.stop * n)
-            labels_slice = [label[new_slice] for label in labels]
+            # special case for ctc loss
+            if len(labels) > 0 and len(labels[0].shape) == 2 \
+               and labels[0].shape[0] == (islice.stop - islice.start) \
```

Comment:
I add labels[0].shape[0] == (islice.stop - islice.start) to make sure this is for ctc case.

Here, (islice.stop - islice.start) = batch_size.